### DISCUSSION TITLE
Add basic instructions for writing custom filter functions to docs

### DISCUSSION THREAD

**Discussion by Saithe6** [awesomeWM/awesome#3952] [Ideas]
Atm it's very difficult to figure out how to write a custom function for filtering the taglist (I think the tasklist might have filtering too? Idk I don't use it) when it's ultimately pretty simple. For those who're wondering, the function has to take an argument that represents the tag and return true if it should be shown or false if it should be hidden.

**Comment by Aire-One** [comment]
The doc from the git version is a more detailed and includes a description of the callback constraints: https://awesomewm.org/apidoc/widgets/awful.widget.taglist.html#filter. Ultimately, this is a pretty common pattern for a filter function

**Comment by gabrielfrimodig** [comment]
As @Aire-One mentioned, the [documentation](https://awesomewm.org/apidoc/widgets/awful.widget.taglist.html#filter) from git version is quite detailed.

 You can add a custom function to filters the tags like this:
 
 ```lua
local function tag_filter(tag)
    return #tag:clients() > 0
end
```

This function takes a single argument (tag object) and return `true` if the tag should be shown and `false` otherwise. If the tag has clients.

To apply the filter, set the `filter` property:
```lua
my_taglist = awful.widget.taglist {
    screen  = s,
    filter  = tag_filter,
    buttons = taglist_buttons
}
```
